.github/workflows/build-test-deploy.yml

```diff
@@ -137,906 +62,417 @@ jobs:
             **/*.json
             **/*.lock
 
-  check-nix:
-    name: "Check nix"
+  nix-flake-check:
+    name: "Nix flake check"
     needs:
-      - check-privilege
-    continue-on-error: false
+      - privilege-check
     runs-on:
       - self-hosted
       - linux
       - x64
       - sre
```

Comment:
Since this is a lightweight check; we can run this on github infra instead of our own runners.

Reply:
Saves us money + means we need 1 less autoscale.

Reply:
Fixed in 308a340e31f8cfed26a7c7a49d0d50bb8ec75979